code/game/objects/explosion.dm

```diff
@@ -192,9 +192,11 @@
 		//Machines which report explosions.
 		for(var/i,i<=GLOB.doppler_arrays.len,i++)
 			var/obj/machinery/doppler_array/Array = GLOB.doppler_arrays[i]
+			var/obj/item/clothing/head/helmet/space/hardsuit/rd/Helm_Array = GLOB.doppler_arrays[i]
```

Comment:
This is a direct cast, there is no type checking here, so both `Helm_Array` and `Array` will be the same object on each iteration of the loop. Meaning that whatever `GLOB.doppler_arrays[i]` is, will have two calls to `sense_explosion()`.

Reply:
Fixed with your help!